confirmation/migrations/0009_confirmation_expiry_date_backfill.py

```diff
@@ -0,0 +1,67 @@
+# Generated by Django 3.1.7 on 2021-03-31 20:47
+
+import time
+from datetime import timedelta
+
+from django.conf import settings
+from django.db import migrations, transaction
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def set_expiry_date_for_existing_confirmations(
+    apps: StateApps, schema_editor: DatabaseSchemaEditor
+) -> None:
+    Confirmation = apps.get_model("confirmation", "Confirmation")
+    # The values at the time of this migration
+    INVITATION = 2
+    UNSUBSCRIBE = 4
+    MULTIUSE_INVITE = 6
+
+    @transaction.atomic()
+    def backfill_confirmations_between(lower_bound: int, upper_bound: int) -> None:
+        for confirmation in Confirmation.objects.filter(id__gte=lower_bound, id__lte=upper_bound):
+            if confirmation.type in (INVITATION, MULTIUSE_INVITE):
+                confirmation.expiry_date = confirmation.date_sent + timedelta(
+                    settings.INVITATION_LINK_VALIDITY_DAYS
+                )
+            elif confirmation.type == UNSUBSCRIBE:
+                confirmation.expiry_date = confirmation.date_sent + timedelta(1000000)
+            else:
+                confirmation.expiry_date = confirmation.date_sent + timedelta(
+                    settings.CONFIRMATION_LINK_DEFAULT_VALIDITY_DAYS
+                )
+            confirmation.save()
+
+    BATCH_SIZE = 1000
```

Comment:
Is this BATCH_SIZE value fine or do we want something larger?

Reply:
I don't think it's all that necessary to fine-tune this; it's run-once, doesn't block availability while it runs, and accomplishes the important thing of not loading all of them into memory at once.

1000 seems Fine™️.

Reply:
AGreed.